fhir-mapping/src/main/ml-modules/root/data-services/practitioner/search.sjs

```diff
@@ -39,46 +28,63 @@ const codingSystemMap = new Map([
   ['http://terminology.hl7.org/CodeSystem/v2-0203|SS', 'SSN']
 ]);
 
+const fieldToQueryMap = {
+  identifier(values, modifier) {
+    const identifiers = values.map(valueString => {
+      const [value, system] = qu.getValueAndSystemFromString(valueString);
+
+      const search = [];
+
+      if (value !== '') {
+        search.push(cts.jsonPropertyValueQuery('value', value));
+      }
+
+      if (system) {
+        if (modifier !== null && modifier !== 'of-type') { // of-type forces checking the type system instead of the code system
+          throw new Error(`The "${modifier}" modifier is not supported for identifier searches`);
+        }
+
+        // Convert from FHIR system values to persistent data values
+        const systemType = (modifier === 'of-type' ? typeSystemMap : codingSystemMap).get(system);
+        search.push(cts.jsonPropertyValueQuery('key', systemType || system));
+      }
+
+      return search;
+    });
+
+    return cts.jsonPropertyScopeQuery('identifiers', cts.andQuery(identifiers));
+  },
+  given(values, modifier) {
+    return cts.jsonPropertyValueQuery('firstName', values, qu.defaultTextSearchModifiers);
+  },
+  family(values, modifier) {
+    return cts.jsonPropertyValueQuery('lastName', values, qu.defaultTextSearchModifiers);
+  },
+  name(values, modifier) {
+    return cts.jsonPropertyValueQuery(['firstName', 'lastName', 'middleName'], values, qu.defaultTextSearchModifiers);
+  },
+  _id(values, modifier) {
+    return cts.jsonPropertyValueQuery('publicID', values, qu.defaultTextSearchModifiers);
+  },
+  _version(values, modifier) {
+    return cts.jsonPropertyValueQuery('hashValue', values, qu.defaultTextSearchModifiers);
+  },
+  // FUTURE: expand to allow additional timestamp comparison operations?
+  _lastUpdated(values, modifier) {
+    return cts.jsonPropertyRangeQuery('ingestTimestamp', '>=', xs.dateTime(`${values[0]}T00:00:00`));
```

Comment:
This ignores the modifier that the user passes in with the date. While this is a perfect lift of the DS that it is replacing, it is incorrect and should use the modifiers that are provided. {<, <=, =, >=, >} should be supported, for an example of how to do dateTime equality checks with only a supplied date see the location search data service.